pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -322,3 +288,10 @@ func JoinControlPlane(input *ContolPlaneJoinInput) (string, error) {
 func templateBase64Encode(s string) string {
 	return base64.StdEncoding.EncodeToString([]byte(s))
 }
+
+func templateYAMLIndent(i int, input string) string {
+	split := strings.Split(input, "\n")
+	ident := "\n" + strings.Repeat(" ", i)
+	// Don't indent the first line, it's already indented in the template
```

Comment:
Seems a little flaky to rely on the template to be indenting the first line, how about removing the check and just put the template {{}} on the first column?

Reply:
I think it looks uglier in the template, but my care rating is low, 3/10. If you care more than that I'll happily switch it to not assuming the first line is indented.

```
config: |
    {{.EmbeddedConfig | Indent 4}}
```

vs

```
config: |
{{.EmbeddedConfig | Indent 4}}
```

Reply:
I think the latter is totally fine given the `| Ident` call

Reply:
suggest having a unit test for this, so that future changes don't inadvertently break this.